### PR TITLE
Fixed testing docs example to set g

### LINKED ISSUE #3878: Testing docs example for setting 'g' seems incorrect
### Expected Behavior
In the Flask Testing docs: https://flask.palletsprojects.com/en/1.1.x/testing/#faking-resources-and-context

There is an example illustrating how to utilise the @contextmanager to set g early that appears to be misleading. The appcontext_pushed hook isn't correctly invoked following the example given.

In the usage section that uses the test_client, it appears app.app_context() gets the desired results.

Here is how I got it to work:

```python
@contextmanager
def user_set(state, _user):
    def handler(sender, **kwargs):
        g.user = _user

    with appcontext_pushed.connected_to(handler, state):
        yield

# yours is def user_set
with user_set(self.application, user):
    # use app_context() instead of test_client() context manager
    with app.app_context():
        client = app.test_client()
        # then make your request
        ...
```

### Environment

* Python version: 36
* Flask version: 1.1.2
* Werkzeug version: 1.0.1

### PR DESCRIPTION
**Pull request by Eshan-bd**
I updated the testing docs example for setting 'g'.
- fixes #3878

### CODE REVIEW

**Review by georgekald79 — APPROVED** [review]

File: docs/testing.rst [hunk 1]
```diff
@@ -313,7 +313,8 @@ And then to use it::
         return jsonify(username=g.user.username)
 
     with user_set(app, my_user):
-        with app.test_client() as c:
```
Comment (georgekald79) [inline docs/testing.rst:316]:
```suggestion
        with app.test_client() as c:```suggestion
        with app.test_client() as c:
```

```

**Review by georgekald79 — APPROVED** [review]

File: docs/testing.rst (hunk 1 above)
Comment (ThiefMaster) [reply docs/testing.rst:316]: @georgekald79 please stop reviewing random PRs in projects you aren'te ven involved with, especially when the reviews/suggestions are not actually useful.

**Comment by davidism** [comment]
@georgekald79 please do not add review approvals to issues you  are not a maintainer for. They are noisy and do not help us.

**Review by pilven69 — APPROVED** [review]

**Comment by davidism** [comment]
See comment on linked issue. Documented code works as-is from my test.

**Pull request closed without merging**